src/components/DayPicker.jsx

```diff
@@ -96,16 +96,22 @@ export default class DayPicker extends React.Component {
   componentDidMount() {
     if (this.isHorizontal()) {
       this.adjustDayPickerHeight();
-      this.initializeDayPickerWidth();
     }
   }
 
   componentWillReceiveProps(nextProps) {
-    if (!this.hasSetInitialVisibleMonth && !nextProps.hidden) {
-      this.hasSetInitialVisibleMonth = true;
-      this.setState({
-        currentMonth: nextProps.initialVisibleMonth(),
-      });
+    if (!nextProps.hidden) {
```

Comment:
i wonder if perhaps `componentDidUpdate` would be a better place to do this computation, since that should fire every time the component rerenders?